store/mongo/datastore_mongo.go

```diff
@@ -238,15 +243,17 @@ func (db *DataStoreMongo) InsertWorkflows(ctx context.Context, workflows ...mode
 				return i + 1, err
 			}
 		}
-		db.workflows[workflow.Name] = &workflow
+		db.workflows[workflow.Name+"."+strconv.Itoa(workflow.Version)] = &workflow
 	}
 	return len(workflows), nil
 }
 
 // GetWorkflowByName gets the workflow with the given name - either from the
 // cache, or searches the database if the workflow is not cached.
-func (db *DataStoreMongo) GetWorkflowByName(ctx context.Context, workflowName string) (*model.Workflow, error) {
-	workflow, ok := db.workflows[workflowName]
+func (db *DataStoreMongo) GetWorkflowByName(ctx context.Context, workflowName string, version string) (*model.Workflow, error) {
+	workflow, ok := db.workflows[workflowName+"."+version]
```

Comment:
Dummy question: why do you need to store the version number in the key here?
Wouldn't it be enough to keep the implementation as it is, and at the end of the function add the version check logic on the `workflow` variable? I think we don't want to have multiple versions of the workflow in memory, because it increases the complexity and doesn't bring significant advantages.

Reply:
no problema. if we do not want to keep them in memory, then it makes sense. fixed.